airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -332,7 +335,7 @@ private SyncCheckConnectionFailure checkConnections(final GenerateInputActivity.
     final int attemptCreationVersion =
         Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
 
-    if (attemptCreationVersion < CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+    if (attemptCreationVersion < CHECK_BEFORE_SYNC_CURRENT_VERSION || !runCheck) {
```

Comment:
I'm slightly worried this is not temporal backwards compatible. Testing this on `dev` with a long running workflow seems to show this works. @pmossman or @lmossman or @benmoriceau any thoughts?

Reply:
Yeah I don't think this is backwards compatible. If there is some active workflow that ran the check connection, then this change is deployed, and temporal then tries to replay that workflow, and this `runCheck` is false, then temporal will get confused that the check connection activity is not expected in the new temporal logic but that it exists in the replay history

Reply:
would defaulting runCheck to true, and only setting this to false, avoid this issue?

Reply:
I think that would work, though it still opens up the possibility of causing a non-determinism error if someone sets the new env var to true. 

In order to make this fully backwards compatible, you could add another version check, e.g. something like `SKIP_CHECK_CONNECTION_VERSION` and check that before checking `runCheck`, so that workflows that were started before this change are guaranteed to be consistent. It's slightly uglier, but safer. And we should hopefully be able to remove all of these version checks together later on

Reply:
Can we move the check at the activity level and rename the variable something like `overrideCheckConnection`. If the env is true, the activity will always return that the check worked, if false it will run the check. That will allow to avoid any version issues.

It is true that setting the default to true will work but it is not the best solution because it can be activated outside of the integration tests and adding a new version won't work either because the env can change without having a new version.